Sources/Logging/Logging.swift

```diff
@@ -39,20 +39,44 @@ import WASILibc
 /// logger.info("Hello World!")
 /// ```
 public struct Logger {
-    @usableFromInline
-    var handler: LogHandler
+    /// A private property to hold the boxed `LogHandler`.
+    private var _handler: Storage
 
     /// An identifier of the creator of this `Logger`.
     public let label: String
+    
+    /// A computed property to access the `LogHandler`.
+    public var handler: LogHandler {
+        get {
+            return _handler.handler
+        }
+        set {
+            if !isKnownUniquelyReferenced(&self._handler) {
+                self._handler = Storage(label: self.label, handler: newValue)
+            } else {
+                self._handler.handler = newValue
+            }
+        }
+    }
 
     /// The metadata provider this logger was created with.
     public var metadataProvider: Logger.MetadataProvider? {
-        return self.handler.metadataProvider
+        return handler.metadataProvider
     }
 
     internal init(label: String, _ handler: LogHandler) {
         self.label = label
-        self.handler = handler
+        self._handler = Storage(label: label, handler: handler)
+    }
+
+    public class Storage: @unchecked Sendable {
```

Comment:
The below should hopefully fix this. Can we also move the class to top of this struct. So that it is the first line of code inside the structs body.
```suggestion
    @usableFromInline
    public final class Storage: @unchecked Sendable {
```

Reply:
This needs to be `internal class Storage` and then you also don't need the `final` but I would still recommend putting the `final`.

Also: A comment is warranted: `Storage` isn't actually `Sendable` but the compiler can't understand that we're using it with copy-on write... So it's never mutated unless we're making a copy. This is safe but compiler doesn't know. This makes it extra crucial that this is _not_ public, or else the `Sendable` conformance would be wrong.

I would write

```
internal final class Storage: @unchecked /* and not actually */ Sendable /* but safe if only used with CoW */ {
 ...
}
```

Reply:
@weissi The [log](https://ci.swiftserver.group/job/swift-log-swift51-prb/351/console) is failing with the below errors
 error: unknown attribute 'unchecked'
error: use of undeclared type 'Sendable'
Do I need to to conform Storage to Sendable? (I was having warnings without that)

Reply:
> error: unknown attribute 'unchecked'
> error: use of undeclared type 'Sendable'

That means the compiler is too old. Ignore any failures that appear in Swift 5.0 through 5.7. We only care about Swift 5.8, 5.9, 5.10 and Swift 6 at this point. Those versions do have `Sendable` and `@unchecked`.

Reply:
@ayushi2103 can you add the conformance to Sendable by putting it into a `#if compiler(` directive. There should be other examples in this repo that use `#if`

Reply:
@FranzBusch we don't care about 5.0...5.7 anymore, so I don't think we should `#if` things that work on 5.8+, right?